agent.go

```diff
@@ -229,18 +235,20 @@ func (agent *agentS) head(url string) (string, error) {
 }
 
 func (agent *agentS) request(url string, method string, data interface{}) (string, error) {
-	return agent.fullRequestResponse(url, method, data, nil, "")
+	return agent.fullRequestResponse(nil, url, method, data, nil, "")
```

Comment:
Why not pass `context.Background()` instead of `nil`?

Reply:
Actually no reason for that. It will be set inside a function anyway, but if looks better, I will add it her 👍

Reply:
I believe it's also a convention to not to pass `nil` context and use `context.Background()`/`context.TODO()` instead.